### DISCUSSION TITLE
How to get linestretch option to not impact table generation?

### DISCUSSION THREAD

**Discussion by victorfeagins** [quarto-dev/quarto-cli#9128] [Q&A] [answered]
### Description

I would like to include tables in my pdf document that has different line spacing compared to rest of the document. 

I noticed while writing a report that quarto linestretch will override flextable's `line_spacing` command. I can't tell if this is a bug or if there is a combination of commands that will make flextable `line_spacing` to work correctly or if we can make a local environment that has a custom line spacing. I tired the results: asis to get around the formatting but that did not work. I provided a simple example:

````qmd
---
title: "Untitled"
format: 
  pdf:
    linestretch: 2
---

```{r}
library(flextable)
```
```{r}
#| tbl-cap: This is a table
dat<- iris[1:5,]
flextable(dat) |> 
  line_spacing(space = .5)
```


```{r}
#| tbl-cap: This is a table
#| results: asis
dat<- iris[1:5,]
flextable(dat) |> 
  line_spacing(space = .5) |> 
  flextable_to_rmd()

```
````
I am on quarto verison 1.4.534, Windows 11.

In the meantime, I save the flextable separately and import them as high quality png in my reports.

**Comment by mcanouil** [comment]
You can't (pretty sure but not 100 %).
That's LaTeX limitation I believe.
You'll have better luck looking for workaround/hack on places more dedicated to LaTeX.

**Comment by victorfeagins** [comment, marked as answer]
Using the setstretch command it will work but will impact the spacing of the code block as well. Which is fine for my needs.  I tried manipulating the line-height directly with CSS and that did not work if I did it correctly.

````qmd
---
title: "Untitled"
format: 
  pdf:
    linestretch: 2
engine: knitr
---

```{css}
.table_format {
  line-height: .5;
}   
```

```{r}
library(flextable)
```
```{r}
#| tbl-cap: This is a table
dat<- iris[1:5,]
flextable(dat)
```


\setstretch{1.0}
```{r}
#| tbl-cap: This is a table
flextable(dat)
```

\setstretch{2}

```{r}
#| class-output: table_format
#| tbl-cap: This is a table
dat<- iris[1:5,]
flextable(dat)
```

````

**Reply by mcanouil** [reply]
LaTeX has nothing to do with HTML/CSS.
It's a completely different and independent technology/language.

Linestretch is a page setup in LaTeX. It applies on every lines.
A table has lines (rows).
The same thing would happen in Word if you set line spacing, any content will abide by this setting.

You should raw code block/inline syntax to use LaTeX.
- `` `whatever LaTeX`{=latex}``
- `` ```{=latex}``
